gsa/src/web/pages/tasks/listpage.js

```diff
@@ -156,15 +166,29 @@ const TasksListPage = () => {
     [deleteTask, refetch, showError],
   );
 
+  const handleBulkDeleteTask = () => {
+    let tasksToDelete;
+    if (selectionType === SelectionType.SELECTION_USER) {
+      tasksToDelete = getEntityIds(selected);
+    } else if (selectionType === SelectionType.SELECTION_PAGE_CONTENTS) {
+      tasksToDelete = getEntityIds(tasks);
+    } else {
+      tasksToDelete = getEntityIds(allFilteredTasks);
+    }
+    return deleteTasks(tasksToDelete).then(refetch, showError);
+  };
+
   useEffect(() => {
     // load tasks initially after the filter is resolved
     if (!isLoadingFilter && hasValue(filter) && !called) {
       getTasks({
         filterString: filter.toFilterString(),
         first: filter.get('rows'),
       });
+
+      getAllFiltered({filterString: filter.all().toFilterString()});
```

Comment:
Oh just saw this line. This isn't a good idea. It will always load **ALL** tasks.

Reply:
We need to load all tasks only on demand. And only if the selection was filter related. Maybe it is better to implement a deleteAllTasks graphql API to make this easier?

Reply:
I thought this just sends the current filter with rows=-1?

Reply:
I'll try this with a useEffect hook. I tried for hours to fetch this only in case 3 of `handleBulkDeleteTask` but don't think the data can be used in the same render, or something.